### DISCUSSION TITLE
"Code-splitting" for production builds

### DISCUSSION THREAD

**Discussion by sumanthratna** [alpinejs/alpine#678] [4. General]
Note: this is on the order of a micro-optimization since Alpine is already so small. However, it's nice knowing that no unnecessary JS is being sent over a network call. 

# Feature Request
Along with uploading `alpine.min.js` to CDNs, also upload `alpine.x-data.min.js`, `alpine.x-init.min.js`, `alpine.x-show.min.js`, etc. This way, if I'm only using the `x-data` directive, I don't need to worry about unnecessarily loading `x-show`. 

I completely understand if this is too much of a hassle. Alpine is great!

**Comment by HugoDF** [comment]
I did some work to generate a "lite" build (pretty much just x-show,x-bind,x-on) but we decided against merging it because like you said, Alpine is already pretty small.

I think this might be on the roadmap for v3 whereby each directive is just added to Alpine so we can do what you're saying: include Alpine core, then each extra script adds specific functionality
